### PR TITLE
[MINOR][K8S] add missing docs for podTemplateContainerName properties

### PR DESCRIPTION
**Pull request by aditanase**
## What changes were proposed in this pull request?

Adding docs for an enhancement that came in late in this PR: #22146 
Currently the docs state that we're going to use the first container in a pod template, which was the implementation for some time, until it was improved with 2 new properties.

## How was this patch tested?

I tested that the properties work by combining pod templates with client-mode and a simple pod template.

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by aditanase** [comment]
cc @mccheah @onursatici @skonto

**Comment by dongjoon-hyun** [comment]
Thank you for your first contribution, @aditanase .

**Comment by dongjoon-hyun** [comment]
ok to test

File: docs/running-on-kubernetes.md [hunk 1]
```diff
@@ -932,15 +935,31 @@ specific to Spark on Kubernetes.
...
   <td><code>spark.kubernetes.executor.podTemplateFile</code></td>
   <td>(none)</td>
   <td>
    Specify the local file that contains the executor [pod template](#pod-template). For example
-   <code>spark.kubernetes.executor.podTemplateFile=/path/to/executor-pod-template.yaml`</code>
+   <code>spark.kubernetes.executor.podTemplateFile=/path/to/executor-pod-template.yaml</code>
+  </td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.driver.podTemplateContainerName</code></td>
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md]: Could you move this to line 941 after `spark.kubernetes.driver.podTemplateFile` config? We had better group together `spark.kubernetes.driver.*` configs and `spark.kubernetes.executor.*` configs.

Comment (aditanase) [reply docs/running-on-kubernetes.md]: Thanks, done. I'll squash the commits before merging.

**Comment by aditanase** [comment]
@dongjoon-hyun is this ok to merge? Should I squash it and ping you again?

File: docs/running-on-kubernetes.md [hunk 2]
```diff
@@ -932,15 +935,31 @@ specific to Spark on Kubernetes.
...
    Specify the local file that contains the driver [pod template](#pod-template). For example
-   <code>spark.kubernetes.driver.podTemplateFile=/path/to/driver-pod-template.yaml`</code>
+   <code>spark.kubernetes.driver.podTemplateFile=/path/to/driver-pod-template.yaml</code>
+  </td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.driver.podTemplateContainerName</code></td>
+  <td>(none)</td>
+  <td>
+   Specify the container name to be used as a basis for the driver in the given [pod template](#pod-template).
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md]: Actually, `[pod template](#pod-template)` is not working. Could you try the following and check with `SKIP_API=1 jekyll build`?
```
<a href="#pod-template">pod template</a>
```

File: docs/running-on-kubernetes.md [hunk 3]
```diff
@@ -932,15 +935,31 @@ specific to Spark on Kubernetes.
   <td>(none)</td>
   <td>
    Specify the local file that contains the driver [pod template](#pod-template). For example
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md]: Please fix this, too.

File: docs/running-on-kubernetes.md [hunk 4]
```diff
@@ -932,15 +935,31 @@ specific to Spark on Kubernetes.
...
+  <td>
+   Specify the container name to be used as a basis for the driver in the given [pod template](#pod-template).
+   For example <code>spark.kubernetes.driver.podTemplateContainerName=spark-driver</code>
   </td>
 </tr>
 <tr>
   <td><code>spark.kubernetes.executor.podTemplateFile</code></td>
   <td>(none)</td>
   <td>
    Specify the local file that contains the executor [pod template](#pod-template). For example
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md]: ditto.

File: docs/running-on-kubernetes.md [hunk 5]
```diff
@@ -932,15 +935,31 @@ specific to Spark on Kubernetes.
...
-   <code>spark.kubernetes.executor.podTemplateFile=/path/to/executor-pod-template.yaml`</code>
+   <code>spark.kubernetes.executor.podTemplateFile=/path/to/executor-pod-template.yaml</code>
+  </td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.executor.podTemplateContainerName</code></td>
+  <td>(none)</td>
+  <td>
+   Specify the container name to be used as a basis for the executor in the given [pod template](#pod-template).
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md]: ditto.

**Comment by srowen** [comment]
Ping @aditanase

**Comment by aditanase** [comment]
@dongjoon-hyun Thanks for catching the broken anchors. Fixed them per your suggestion and tested with a jekyll build. Should work now.

@srowen my apologies, I've been on an extended winter break. I've rebased against master and squashed my commits - should be ready to merge.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**